napari/_qt/dialogs/_tests/test_reader_dialog.py

```diff
@@ -126,6 +128,47 @@ def _(path):
     assert tmp_plugin.name not in readers
 
 
+def test_open_sample_data_shows_all_readers(
+    make_napari_viewer,
+    tmp_plugin: DynamicPlugin,
+):
+    """Checks that sample data callback `_add_sample` shows all readers."""
+    tmp2 = tmp_plugin.spawn(register=True)
+
+    @tmp_plugin.contribute.reader(filename_patterns=['*.fake'])
+    def _(path):
+        ...
+
+    @tmp2.contribute.reader(filename_patterns=['*.fake'])
+    def _(path):
+        ...
+
+    my_sample = SampleDataURI(
+        key='tmp-sample',
+        display_name='Temp Sample',
+        uri='some-path/some-file.fake',
+    )
+    tmp_plugin.manifest.contributions.sample_data = [my_sample]
+    from napari._qt.dialogs.qt_reader_dialog import QtReaderDialog
+
+    viewer = make_napari_viewer()
+    sample_action = viewer.window.file_menu.open_sample_menu.actions()[0]
+
+    with (
+        mock.patch(
+            'napari._qt.dialogs.qt_reader_dialog.prepare_remaining_readers'
+        ) as mock_prepare_readers,
+        mock.patch.object(
+            QtReaderDialog, 'get_user_choices', return_value=(None, None)
```

Comment:
This had to be mocked because a otherwise a dialogue would pop open and stay open during test.